### PR TITLE
sink/kafka(ticdc): support PLAIN and GSSAPI auth

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4423

### What is changed and how it works?

- Support PLAIN auth
- Support GSSAPI auth

I refer to the check logic here at https://github.com/Shopify/sarama/blob/947343309601b4eb3c2fa3e7d15d701b503dd491/config.go#L563 to fill in these values. I didn't add a duplicate check because we create the sink directly when we create the changefeed, so it will be checked.

### Check List 

Tests 

 - [x] Unit test
 - [x] Manual test (add detailed scripts or steps below)
  - https://github.com/pingcap/tiflow/pull/5122#issuecomment-1091382426
  - https://github.com/pingcap/tiflow/pull/5122#issuecomment-1092413151

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

None

Related changes

- [ ] Needs update docs

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
GSSAPI:
1. start kafka playgroud with https://github.com/hi-rustin/kafka-docker-playground/tree/master/environment/kerberos
2. cp kafka-sarama.key form broker container into `configs`
3. create changefeed:
```sh
./cdc cli changefeed create --pd http://upstream-pd:2379 --log-level debug  --sink-uri="kafka://broker.kerberos-demo.local:9092/connect-offsets?protocol=open-protocol&sasl-mechani
sm=gssapi&sasl-gssapi-auth-type=keytab&sasl-gssapi-keytab-path=/configs/kafka-sarama.key&sasl-gssapi-kerberos-config-path=/configs/krb5.conf&sasl-gssapi-service-name=kafka&sasl-gssapi
-user=sarama&sasl-gssapi-realm=TEST.CONFLUENT.IO&sasl-gssapi-disable-pafxfast=false&replication-factor=1&partition-num=1" -c "test"
```
4. created
```log
[2022/04/08 03:41:46.096 +00:00] [INFO] [kafka.go:239] ["async client closed"] [duration=595.285µs] [changefeed=] [role=cdc-client]
[2022/04/08 03:41:46.097 +00:00] [DEBUG] [async_producer.go:1111] ["Producer shutting down."] [name=sarama]
[2022/04/08 03:41:46.098 +00:00] [INFO] [kafka.go:249] ["sync client closed"] [duration=957.626µs] [changefeed=] [role=cdc-client]
[2022/04/08 03:41:46.097 +00:00] [DEBUG] [sarama.go:122] ["Closed connection to broker broker.kerberos-demo.local:9092"] [name=sarama]
[2022/04/08 03:41:46.099 +00:00] [DEBUG] [sarama.go:125] ["Closing Client"] [name=sarama]
[2022/04/08 03:41:46.100 +00:00] [INFO] [kafka.go:260] ["kafka cluster admin closed"] [duration=1.741042ms] [changefeed=] [role=cdc-client]
{"level":"debug","ts":"2022-04-08T03:41:46.104Z","caller":"v3@v3.5.2/retry_interceptor.go:53","msg":"retrying of unary invoker","target":"etcd-endpoints://0xc0008e8700/upstream-pd:2379","attempt":0}
[2022/04/08 03:41:46.101 +00:00] [DEBUG] [sarama.go:122] ["Closed connection to broker broker2.kerberos-demo.local:9092"] [name=sarama]
[2022/04/08 03:41:46.105 +00:00] [DEBUG] [sarama.go:122] ["Closed connection to broker broker.kerberos-demo.local:9092"] [name=sarama]
[2022/04/08 03:41:46.105 +00:00] [DEBUG] [sarama.go:122] ["Closed connection to broker broker.kerberos-demo.local:9092"] [name=sarama]
Create changefeed successfully!
ID: test
Info: {"sink-uri":"kafka://broker.kerberos-demo.local:9092/connect-offsets?protocol=open-protocol\u0026sasl-mechanism=gssapi\u0026sasl-gssapi-auth-type=keytab\u0026sasl-gssapi-keytab-path=/configs/kafka-sarama.key\u0026sasl-gssapi-kerberos-config-path=/configs/krb5.conf\u0026sasl-gssapi-service-name=kafka\u0026sasl-gssapi-user=sarama\u0026sasl-gssapi-realm=TEST.CONFLUENT.IO\u0026sasl-gssapi-disable-pafxfast=false\u0026replication-factor=1\u0026partition-num=1","opts":{},"create-time":"2022-04-08T03:41:44.782689125Z","start-ts":432377509774884865,"target-ts":0,"admin-job-type":0,"sort-engine":"unified","sort-dir":"","config":{"case-sensitive":true,"enable-old-value":true,"force-replicate":false,"check-gc-safe-point":true,"filter":{"rules":["*.*"],"ignore-txn-start-ts":null},"mounter":{"worker-num":16},"sink":{"dispatchers":null,"protocol":"open-protocol","column-selectors":null},"cyclic-replication":{"enable":false,"replica-id":0,"filter-replica-ids":null,"id-buckets":0,"sync-ddl":false},"scheduler":{"type":"table-number","polling-time":-1},"consistent":{"level":"none","max-log-size":64,"flush-interval":1000,"storage":""}},"state":"normal","error":null,"sync-point-enabled":false,"sync-point-interval":600000000000,"creator-version":"v6.0.0-master-dirty"}
```

**Comment by hi-rustin** [comment]
@3AceShowHand @amyangfei ping~

File: pkg/security/sasl.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	// SCRAM256Mechanism means the SASL mechanism is SCRAM-SHA-256.
+	SCRAM256Mechanism SASLMechanism = sarama.SASLTypeSCRAMSHA256
+	// SCRAM512Mechanism means the SASL mechanism is SCRAM-SHA-512.
+	SCRAM512Mechanism SASLMechanism = sarama.SASLTypeSCRAMSHA512
+	// GSSAPIMechanism means the SASL mechanism is GSSAPI.
+	GSSAPIMechanism SASLMechanism = sarama.SASLTypeGSSAPI
+)
+
+// SASLMechanismFormString converts the string to SASL mechanism.
+func SASLMechanismFormString(s string) (SASLMechanism, error) {
```
Comment (amyangfei) [inline pkg/security/sasl.go:41]: s/Form/From

File: pkg/security/sasl_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+package security
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSASLMechanismFormString(t *testing.T) {
```
Comment (amyangfei) [inline pkg/security/sasl_test.go:22]: s/Form/From

**Review by amyangfei — COMMENTED** [review]
Please remember to update docs repo about the SASL mechanism changes, including
- old version TiCDC doesn't support PLAIN mechanism
- PLAIN and GSSAPI auth are added since version (maybe v6.1)?

File: pkg/security/sasl.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/security/sasl.go]: Good catch! 
Thanks for your review!  💚 💙 💜 💛 ❤️

**Review by amyangfei — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**